src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray`1+Builder.cs

```diff
@@ -304,14 +296,13 @@ public void AddRange(T[] items, int length)
                 Requires.NotNull(items, "items");
                 Requires.Range(length >= 0, "length");
 
+                if (length > items.Length)
+                    throw new IndexOutOfRangeException("length");
```

Comment:
The parameter to the ctor is a message rather than a parameter name; this should probably just use the parameterless ctor and let it use the default message.  I'd suggest also adding a comment to indicate this is being done for backwards compat.

This is, of course, assuming that we care about this exception not changing.  If we're ok with it changing, since it probably should have been an argument exception in the first place, then we could instead do argument validation with Requires.Range and remove this (tests would also need to be updated, as there are tests validating this behavior).  @terrajobst?  @AArnott?